skimage/color/colorconv.py

```diff
@@ -824,8 +824,12 @@ def gray2rgb(image, alpha=None):
     Raises
     ------
     ValueError
-        If the input is not a 2- or 3-dimensional image.
+        If the input is not a 1-, 2- or 3-dimensional image.
 
+    Notes
+    -----
+    If the input is a 1-dimensional image of shape ``(M, )``, the output
+    will be shape ``(M, 1, 3)``.
```

Comment:
@Buntworthy yes, the confusion with a 1D image is a problem, which is why I would be mostly ok with `(1, M, 3)`. But as you say, there is potential for confusion either way, and at any rate, that's why we are heading towards the convention of requiring a `multichannel=True` kwarg for any function that can handle both color and grayscale images, to avoid such ambiguity.

So, yes, go ahead and make the change to (M, 3). I agree that higher-D is out of scope. =)